### PR TITLE
feat: add edge commands 

### PR DESCRIPTION
**Pull request by fraidev**
Needs:
- [x] https://github.com/infinyon/fluvio/pull/3931

Implementation of edge commands:

```
fluvio edge status
```

```
flvd edge connect --file edge.json
```

### CODE REVIEW

File: crates/fluvio-cli/src/client/edge/status.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                    }
+                    _ => None,
+                }
+            })
+            .collect();
+        output::format(out, outlist, self.output.format)
+    }
+}
+
+#[allow(dead_code)]
```
Comment (galibey) [inline crates/fluvio-cli/src/client/edge/status.rs:46]:
```suggestion
#[allow(dead_code)]
```

File: crates/fluvio-cli/src/client/edge/status.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+mod output {
+
+    //!
+    //! # Fluvio list - output processing
+    //!
+    //! Format SmartModules response based on output type
```
Comment (galibey) [inline crates/fluvio-cli/src/client/edge/status.rs:52]:
```suggestion
    //! Format SmartModules response based on output type
```

File: crates/fluvio-cli/src/client/edge/status.rs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    type ListVec = Vec<(String, String, String, String)>;
+
+    #[derive(Serialize)]
+    struct TableList(ListVec);
+
+    // -----------------------------------
+    // Format Output
+    // -----------------------------------
+
+    /// Format SmartModules based on output type
```
Comment (galibey) [inline crates/fluvio-cli/src/client/edge/status.rs:73]:
```suggestion
    /// Format SmartModules based on output type
```

**Review by galibey — APPROVED** [review]
LGTM

File: crates/fluvio-cli/src/client/edge/status.rs (hunk 1 above)
Comment (fraidev) [reply crates/fluvio-cli/src/client/edge/status.rs]: Fixed

File: crates/fluvio-cli/src/client/edge/status.rs (hunk 2 above)
Comment (fraidev) [reply crates/fluvio-cli/src/client/edge/status.rs]: Fixed

File: crates/fluvio-cli/src/client/edge/status.rs (hunk 3 above)
Comment (fraidev) [reply crates/fluvio-cli/src/client/edge/status.rs]: Fixed

**Pull request merged**